kernel/kernel-api/src/test/scala/io/delta/kernel/internal/icebergcompat/IcebergUniversalFormatMetadataValidatorAndUpdaterSuite.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.icebergcompat
+
+import scala.collection.JavaConverters._
+
+import io.delta.kernel.exceptions.KernelException
+import io.delta.kernel.internal.TableConfig
+import io.delta.kernel.internal.actions.Metadata
+import io.delta.kernel.internal.util.ColumnMappingSuiteBase
+import io.delta.kernel.types.IntegerType
+import io.delta.kernel.types.StructType
+
+import org.scalatest.funsuite.AnyFunSuiteLike
+
+class IcebergUniversalFormatMetadataValidatorAndUpdaterSuite extends AnyFunSuiteLike
+    with ColumnMappingSuiteBase {
+  test("validateAndUpdate should return empty when UNIVERSAL_FORMAT_ENABLED_FORMATS is not set") {
+    val metadata = createMetadata(Map("unrelated_key" -> "unrelated_value"))
+    val result = IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
+      createMetadata(),
+      metadata)
+    assert(!result.isPresent)
+  }
+
+  test(
+    "validateAndUpdate should return empty when no iceberg in UNIVERSAL_FORMAT_ENABLED_FORMATS") {
+    val metadata = createMetadata(Map(
+      TableConfig.UNIVERSAL_FORMAT_ENABLED_FORMATS.getKey -> "hudi",
+      "unrelated_key" -> "unrelated_value"))
+    val result = IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
+      createMetadata(),
+      metadata)
+    assert(!result.isPresent)
+  }
+
+  test("validateAndUpdate should return empty when iceberg is enabled" +
```

Comment:
Maybe these tests in general could be structured or at least named to be more like a matrix of (icebergCompatV2WasEnabled, icebergCompatV2IsEnabled) since it's essentially a combo of oldMetadata x newMetadata?

Reply:
Tests have been simplified, please let me know structure is still an issue.